pkg/generate/app/cmd/newapp.go

```diff
@@ -312,17 +314,19 @@ func (c *AppConfig) componentsForRepos(repositories app.SourceRepositories) (app
 			errs = append(errs, fmt.Errorf("source not detected for repository %q", repo))
 			continue
 		case info.Dockerfile != nil && (len(c.Strategy) == 0 || c.Strategy == "docker"):
-			df, err := info.Dockerfile.Dockerfile()
-			if err != nil {
-				// an unparseable docker file should not terminate the entire create - it may be valid to Docker but not to us
-				glog.V(1).Infof("The Dockerfile for the repository %q could not be parsed - no image stream can be created for the FROM", info.Path)
+			node := info.Dockerfile.AST()
+			indices := dockerfileutil.FindAll(node, command.From)
+			if len(indices) == 0 {
+				errs = append(errs, fmt.Errorf("the Dockerfile for the repository %q has no FROM instruction", info.Path))
 				continue
 			}
-			dockerFrom, ok := df.GetDirective("FROM")
-			if !ok || len(dockerFrom) > 1 {
-				errs = append(errs, fmt.Errorf("invalid FROM directive in Dockerfile in repository %q", repo))
+			lastFrom := node.Children[indices[len(indices)-1]]
```

Comment:
Here we're correctly looking for the last FROM instruction.
Earlier we were looking at the first FROM, which was a bug.